mixer/pkg/attribute/envoyProtoBag.go

```diff
@@ -0,0 +1,316 @@
+
+	return pb
+}
+
+// GetProtoBag returns a proto-based attribute bag.
+// When you are done using the proto bag, call the Done method to recycle it.
+func GetEnvoyProtoBagAccessLog(msg *accessLogGRPC.StreamAccessLogsMessage, num int) *EnvoyProtoBag {
+	// build the message-level dictionary
+	reqMap := make(map[string]interface{})
+	if httpLogs := msg.GetHttpLogs(); httpLogs != nil {
+		reqMap["context.protocol"] = ""
+		reqMap["destination.ip"] = []byte(net.ParseIP(httpLogs.GetLogEntry()[num].GetCommonProperties().
+			GetDownstreamLocalAddress().GetSocketAddress().GetAddress()).To16())
+		reqMap["destination.port"] = int64(httpLogs.GetLogEntry()[num].GetCommonProperties().
+			GetDownstreamLocalAddress().GetSocketAddress().GetPortValue())
+		reqMap["source.ip"] = []byte(net.ParseIP(httpLogs.GetLogEntry()[num].GetCommonProperties().
+			GetDownstreamRemoteAddress().GetSocketAddress().GetAddress()).To16())
+		reqMap["source.port"] = int64(httpLogs.GetLogEntry()[num].GetCommonProperties().
+			GetDownstreamRemoteAddress().GetSocketAddress().GetPortValue())
+		reqMap["source.namespace"] = httpLogs.GetLogEntry()[num].GetCommonProperties().GetMetadata().GetFilterMetadata()["istio_authn"].GetFields()["source.namespace"].GetStringValue()
+		reqMap["source.principal"] = httpLogs.GetLogEntry()[num].GetCommonProperties().GetMetadata().GetFilterMetadata(
+		)["istio_authn"].GetFields()["source.principal"].GetStringValue()
+		reqMap["source.user"] = httpLogs.GetLogEntry()[num].GetCommonProperties().GetMetadata().GetFilterMetadata()["istio_authn"].GetFields()["source.user"].GetStringValue()
+		reqMap["request.headers"] = httpLogs.GetLogEntry()[num].GetRequest().GetRequestHeaders()
+		reqMap["request.host"] = httpLogs.GetLogEntry()[num].GetRequest().GetAuthority()
+		reqMap["request.method"] = httpLogs.GetLogEntry()[num].GetRequest().GetRequestMethod()
+		reqMap["request.path"] = httpLogs.GetLogEntry()[num].GetRequest().GetPath()
+		reqMap["request.scheme"] = httpLogs.GetLogEntry()[num].GetRequest().GetScheme()
+		reqMap["request.time"] = time.Unix(int64(httpLogs.GetLogEntry()[num].GetCommonProperties().GetStartTime().Seconds),
+			int64(httpLogs.GetLogEntry()[num].GetCommonProperties().GetStartTime().Nanos))
+		reqMap["request.url_path"] = ""
+		reqMap["request.useragent"] = httpLogs.GetLogEntry()[num].GetRequest().GetUserAgent()
+		reqMap["response.headers"] = httpLogs.GetLogEntry()[num].GetResponse().GetResponseHeaders()
+		reqMap["response.code"] = httpLogs.GetLogEntry()[num].GetResponse().GetResponseCode().GetValue()
+		reqMap["response.size"] = httpLogs.GetLogEntry()[num].GetResponse().GetResponseBodyBytes()
+		reqMap["response.total_size"] = httpLogs.GetLogEntry()[num].GetResponse().
+			GetResponseBodyBytes() + httpLogs.GetLogEntry()[num].GetResponse().GetResponseHeadersBytes()
+		reqMap["context.reporter.kind"] = ""
+
+
+	} else if tcpLogs := msg.GetTcpLogs(); tcpLogs != nil {
+		reqMap["context.protocol"] = ""
```

Comment:
for this, i think we can definitely say `tcp`

Reply:
done